clients/src/main/java/com/compliance/financialserviceprovider/controllers/ClaimTemplateController.java

```diff
@@ -92,30 +91,35 @@ private List<ClaimTemplateSuggestion> getSuggestionByLinearId(@PathVariable Stri
-        proxy.startTrackedFlowDynamic(
-                CreateClaimTemplateSuggestion.CreateClaimTemplateSuggestionInitiator.class,
-                claimTemplateSuggestionDTO.getName(),
-                claimTemplateSuggestionDTO.getTemplateDescription(),
-                UniqueIdentifier.Companion.fromString(claimTemplateSuggestionDTO.getRule()),
-                new Date()
-        ).getReturnValue().get();
+        Set<Party> partySet = proxy.partiesFromName("Supervisory Authority", true);
 
-        List<ClaimTemplateSuggestion> claimTemplateSuggestions = proxy
-                .vaultQuery(ClaimTemplateSuggestion.class)
-                .getStates()
-                .stream()
-                .map(
-                        claimTemplateSuggestionStateAndRef -> claimTemplateSuggestionStateAndRef.getState().getData())
-                .collect(Collectors.toList());
+        if (!partySet.isEmpty()) {
+            Party supervisoryAuthority = new ArrayList<>(partySet).get(0);
 
-        // Return regulation linear ID
-        return claimTemplateSuggestions
-                .stream()
-                .filter(
-                        claimTemplateSuggestion -> claimTemplateSuggestion.getName().equals(claimTemplateSuggestionDTO.getName()) && claimTemplateSuggestion.getTemplateDescription().equals(claimTemplateSuggestionDTO.getTemplateDescription())
-                )
-                .collect(Collectors.toList())
-                .get(0);
+            proxy.startTrackedFlowDynamic(
+                    CreateClaimTemplateSuggestion.CreateClaimTemplateSuggestionInitiator.class,
+                    claimTemplateSuggestionDTO.getName(),
+                    claimTemplateSuggestionDTO.getTemplateDescription(),
+                    supervisoryAuthority,
+                    UniqueIdentifier.Companion.fromString(claimTemplateSuggestionDTO.getRule())
+                    ).getReturnValue().get();
+
+            List<ClaimTemplateSuggestion> claimTemplateSuggestions = proxy
+                    .vaultQuery(ClaimTemplateSuggestion.class)
+                    .getStates()
+                    .stream()
+                    .map(
+                            claimTemplateSuggestionStateAndRef -> claimTemplateSuggestionStateAndRef.getState().getData())
+                    .collect(Collectors.toList());
 
+            // Return regulation linear ID
+            return claimTemplateSuggestions
+                    .stream()
+                    .filter(
+                            claimTemplateSuggestion -> claimTemplateSuggestion.getName().equals(claimTemplateSuggestionDTO.getName()) && claimTemplateSuggestion.getTemplateDescription().equals(claimTemplateSuggestionDTO.getTemplateDescription())
+                    )
+                    .collect(Collectors.toList())
+                    .get(0);
```

Comment:
Does this return the linear Id? or the complete object

Reply:
It returns the object, the comment is wrong. But this part can throw errors, I already created a bug for it